<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="car1"/>
  <arg name="role" default="leader"/>
  <param name="use_sim_time" value="false"/>  
  
  <group ns="$(arg robot_name)">
    <node name="rosbag_play" pkg="rosbag" type="play" args="--clock $(find obstacle_detector)/rosbag/good.bag"/>
    <node pkg='rviz' type='rviz' name='rviz' args='-d $(find obstacle_detector)/rviz/run_rosbag.rviz'/>
    
    <node pkg="rosservice" type="rosservice" name="Shelf_Detector_Param_set" args="call --wait StringSrv '$(find obstacle_detector)/param/single_AMR.yaml'" />
    <node name="shelf_detector" pkg="obstacle_detector" type="shelf_detector.py" output="screen">
      <!-- System -->
      <param name="robot_name" value="$(arg robot_name)"/>
      <param name="role" value="$(arg role)"/>
      <param name="frequency" value="10"/>
      <!-- Frame ID -->
      <param name="base_link_frame" value="$(arg robot_name)/tmp/base_link"/>
      <param name="base_big_car_frame" value="carB/tmp/base_link"/>
      <!-- Remap -->
      <remap from="search_center" to="tmp/search_center"/>
      <remap from="/$(arg robot_name)/theta" to="/tmp/$(arg robot_name)/theta"/>
      <remap from="obstacle_detector/markers/base" to="obstacle_detector/markers/base/tmp"/>
      <remap from="obstacle_detector/markers/peer" to="obstacle_detector/markers/peer/tmp"/>
    </node>
  </group>
</launch>
